optuna/integration/lightgbm.py

```diff
@@ -72,6 +71,22 @@ def __call__(self, env):
                 raise optuna.structs.TrialPruned(message)
             return None
 
+        is_cv = len(env.evaluation_result_list[0]) == 5
```

Comment:
The code between L75 and L88 seems redundant. How about rewriting this as the following code?
```python
        # If this callback has been passed to `lightgbm.cv` function,
        # the value of `is_cv` becomes `True`. See also:
        # https://github.com/Microsoft/LightGBM/blob/v2.2.2/python-package/lightgbm/engine.py#L329
        is_cv = len(env.evaluation_result_list) > 0 and len(env.evaluation_result_list[0]) == 5
        if is_cv:
            target_valid_name = 'cv_agg'
        else:
            target_valid_name = self.valid_name

        for valid_name, metric, current_score, is_higher_better, *_ in env.evaluation_result_list:
            if valid_name != target_valid_name or metric != self.metric:
                continue

            if is_higher_better:
                raise ValueError(
                    'Pruning using metrics to be maximized has not been supported yet '
                    '(validation_name: {}, metric: {}).'.format(valid_name, metric))

            self.trial.report(current_score, step=env.iteration)
            if self.trial.should_prune(env.iteration):
                message = "Trial was pruned at iteration {}.".format(env.iteration)
                raise optuna.structs.TrialPruned(message)

            return None

        raise ValueError(
            'The entry associated with the validation name "{}" and the metric name "{}" '
            'is not found in the evaluation result list {}.'.format(
                target_valid_name, self.metric, str(env.evaluation_result_list)))
```

Reply:
Thank you for your suggestion. I'm currently modifying my code and found that mypy rejects `*` before `_`, which is included in your suggestion as below.
```python
for valid_name, metric, current_score, is_higher_better, *_ in env.evaluation_result_list:
```

A work around may be 
```python
for evaluation_result in env.evaluation_result_list:
        valid_name, metric, current_score, is_higher_better = evaluation_result[:4]
```

but this seems awkward to me. If you could come up with any other solution, could you help me?

Reply:
Hmm...
It appears that it is a Python3 only feature (cf. [Python star unpacking for version 2.7](https://stackoverflow.com/questions/10792970/python-star-unpacking-for-version-2-7)). Since Optuna supports Python2.7, the feature can not be used...
In my sense, your workaround seems not so bad.

Reply:
Thank you for your help.  I didn't reach out to the page you referred.
I'll use my code thank you!